rest/utilities_testing.go

```diff
@@ -478,3 +494,67 @@ func (s *SlowResponseRecorder) Write(buf []byte) (int, error) {
 
 	return numBytesWritten, err
 }
+
+type BlipTester struct {
+	rt          RestTester
+	sender      *blip.Sender
+	blipContext *blip.Context
+}
+
+func (bt BlipTester) Close() {
+	bt.rt.Close()
+}
+
+func CreateBlipTester(t *testing.T, noConflictsMode bool) (bt BlipTester) {
+
+	EnableBlipSyncLogs()
+
+	// Create an admin handler
+	var adminHandler http.Handler
+	switch noConflictsMode {
+	case true:
+		adminHandler = bt.rt.TestAdminHandlerNoConflictsMode()
+	default:
+		adminHandler = bt.rt.TestAdminHandler()
+
+	}
+
+	// Create a test server and close it when the test is complete
+	srv := httptest.NewServer(adminHandler)
+	defer srv.Close()
+
+	// Construct URL to connect to blipsync target endpoint
+	destUrl := fmt.Sprintf("%s/db/_blipsync", srv.URL)
+	u, err := url.Parse(destUrl)
+	if err != nil {
+		t.Errorf("Error parsing desturl: %v", err)
+	}
+	u.Scheme = "ws"
+
+	// Make BLIP/Websocket connection
+	bt.blipContext = blip.NewContext()
+	bt.blipContext.Logger = func(fmt string, params ...interface{}) {
+		base.LogTo("BLIP", fmt, params...)
+	}
+	bt.blipContext.LogMessages = true
+	bt.blipContext.LogFrames = true
+	origin := "http://localhost" // TODO: what should be used here?
+	bt.sender, err = bt.blipContext.Dial(u.String(), origin)
+	if err != nil {
+		t.Errorf("Websocket connection error: %v", err)
+	}
+
+	return bt
+
+}
+
+func EnableBlipSyncLogs() {
+
+	base.EnableLogKey("HTTP")
+	base.EnableLogKey("HTTP+")
+	base.EnableLogKey("BLIP")
+	base.EnableLogKey("BLIP+")
+	base.EnableLogKey("Sync")
+	base.EnableLogKey("Sync+")
```

Comment:
What's the difference between BLIP and Sync testing?  Does BLIP actually enable logging in go-blip?

Reply:
BLIP has you pass a callback logger in, and so `base.EnableLogKey("BLIP")` enables logging in the callback that's defined in `CreateBlipTester()` 

```
	bt.blipContext.Logger = func(fmt string, params ...interface{}) {
		base.LogTo("BLIP", fmt, params...)
	}
```

Looks like `BLIP+` doesn't do anything from what I can tell, so I'll remove it.